Symfony/CS/Fixer/Contrib/OrderedUseFixer.php

```diff
@@ -13,19 +13,111 @@
  * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ * @author Darius Matulionis <darius@matulionis.lt>
  * @author SpacePossum
  */
 class OrderedUseFixer extends AbstractFixer
 {
     const IMPORT_TYPE_CLASS = 1;
+
     const IMPORT_TYPE_CONST = 2;
+
     const IMPORT_TYPE_FUNCTION = 3;
 
+    const SORT_ALPHA = 'alpha';
+
+    const SORT_LENGTH = 'length';
+
+    /**
+     * Array of import types by which use statements should be ordered. If not defined in configuration only sort algorithm will be applied.
+     *
+     * @var null|int[]
+     */
+    private static $sortTypesOrder = null;
+
+    /**
+     * Array of supported sort types in configuration.
+     *
+     * @var string[]
+     */
+    private static $supportedSortTypes = array(self::IMPORT_TYPE_CLASS, self::IMPORT_TYPE_CONST, self::IMPORT_TYPE_FUNCTION);
+
+    /**
+     * Default sort algorithm.
+     *
+     * @var string
+     */
+    private static $sortAlgorithm = self::SORT_ALPHA;
+
+    /**
+     * Array of supported sort algorithms in configuration.
+     *
+     * @var string[]
+     */
+    private static $supportedSortAlgorithms = array(self::SORT_ALPHA, self::SORT_LENGTH);
+
+    /**
+     * Configuration settings for sort algorithm and sor types.
+     *
+     * array['sortAlgorithm']   string defines sort algorithm.
+     * array['typesOrder']      int[]|null defines import types order or null to sort only by algorithm.
+     *
+     * @param array $configuration
+     */
+    public static function configure(array $configuration = null)
+    {
+        // If no configuration was passed, stick to default.
+        if (null === $configuration) {
+            return;
+        }
+
+        // Check if configuration has all required parameters.
+        if (!array_key_exists('typesOrder', $configuration) || !array_key_exists('sortAlgorithm', $configuration)) {
+            throw new InvalidFixerConfigurationException('ordered_use', sprintf('Configuration array should be composed of: "typesOrder" and "sortAlgorithm".'));
+        }
+
+        /*** Sort types order configuration ***/
+
+        $typesOrder = $configuration['typesOrder'];
+
+        // If no import types order was provided, we will sort only by algorithm.
+        if (null !== $typesOrder) {
+            if (!is_array($typesOrder) || count($typesOrder) !== count(self::$supportedSortTypes)) {
+                throw new InvalidFixerConfigurationException('ordered_use', sprintf('$configuration["typesOrder"] should be array and should be composed of all import types in desired order.'));
+            }
+
+            // Check if all provided sort types are supported.
+            foreach ($typesOrder as $type) {
+                if (!in_array($type, self::$supportedSortTypes, true)) {
+                    throw new InvalidFixerConfigurationException('ordered_use', sprintf('$configuration["typesOrder"] should be array and should be composed of all import types in desired order.'));
```

Comment:
@SpacePossum 

Maby like this:

```php
throw new InvalidFixerConfigurationException('ordered_use', sprintf('Unknown type "%s" in type order configuration, expected all types [ "%s"] to be included in desired order.', $type, implode('","', self::$supportedSortTypes)));
```

because all types should be included.